account_allowance_adjust_transaction_unit_test.go

```diff
@@ -0,0 +1,206 @@
+			Amount:           -1,
+		},
+	}
+	require.Equal(t, expectedHbarAllowances, tx.GetHbarAllowances())
+}
+
+func TestUnitAccountAllowanceAdjustTransactionGrantTokenAllowance(t *testing.T) {
+	tokenID := TokenID{Token: 3}
+	tx:=NewAccountAllowanceAdjustTransaction().
+	GrantTokenAllowance(tokenID, AccountID{Account: 3}, AccountID{Account: 4}, 1)
+	expectedTokenAllowances := []*TokenAllowance{
+		{	
+			TokenID:          &tokenID,
+			SpenderAccountID: &AccountID{Account: 4},
+			OwnerAccountID:   &AccountID{Account: 3},
+			Amount:           1,
+		},
+	}
+	require.Equal(t, expectedTokenAllowances, tx.GetTokenAllowances())
+}
+
+func TestUnitAccountAllowanceAdjustTransactionRevokeTokenAllowance(t *testing.T) {
+	tokenID := TokenID{Token: 3}
+	tx:=NewAccountAllowanceAdjustTransaction().
+	RevokeTokenAllowance(tokenID, AccountID{Account: 3}, AccountID{Account: 4}, 1)
+	expectedTokenAllowances := []*TokenAllowance{
+		{	
+			TokenID:          &tokenID,
+			SpenderAccountID: &AccountID{Account: 4},
+			OwnerAccountID:   &AccountID{Account: 3},
+			Amount:           -1,
+		},
+	}
+	require.Equal(t, expectedTokenAllowances, tx.GetTokenAllowances())
+}
+
+func TestUnitAccountAllowanceAdjustTransactionGrantTokenNftAllowance(t *testing.T) {
+	tokenID := TokenID{Token: 3}
+	NftID := NftID{tokenID, 1}
+	tx:=NewAccountAllowanceAdjustTransaction().
+	GrantTokenNftAllowance(NftID, AccountID{Account: 3}, AccountID{Account: 4})
+	expectedTokenNftAllowances := []*TokenNftAllowance{
+		{
+			SpenderAccountID: &AccountID{Account: 4},
+			OwnerAccountID:   &AccountID{Account: 3},
+			TokenID:          &tokenID,
+			SerialNumbers:   []int64{1},
+			AllSerials:       false,
+			DelegatingSpender: nil,
+		},
+	}
+	require.Equal(t, expectedTokenNftAllowances, tx.GetTokenNftAllowances())
+}
+
+func TestUnitAccountAllowanceAdjustTransactionRevokeTokenNftAllowance(t *testing.T) {
+	tokenID := TokenID{Token: 3}
+	NftID := NftID{tokenID, 1}
+	tx:=NewAccountAllowanceAdjustTransaction().
+	RevokeTokenNftAllowance(NftID, AccountID{Account: 3}, AccountID{Account: 4})
+	expectedTokenNftAllowances := []*TokenNftAllowance{
+		{
+			SpenderAccountID: &AccountID{Account: 4},
+			OwnerAccountID:   &AccountID{Account: 3},
+			TokenID:          &tokenID,
+			SerialNumbers:   []int64{1},
+			AllSerials:       false,
+			DelegatingSpender: nil,
+		},
+	}
+	require.Equal(t, expectedTokenNftAllowances, tx.GetTokenNftAllowances())
+}
+
+func TestUnitAccountAllowanceAdjustTransactionAddAllTokenNftAllowance(t *testing.T) {
+	tokenID := TokenID{Token: 3}
+	tx:=NewAccountAllowanceAdjustTransaction().
+	AddAllTokenNftAllowance(tokenID, AccountID{Account: 3})
+	expectedTokenNftAllowances := []*TokenNftAllowance{
+		{
+			SpenderAccountID: &AccountID{Account: 3},
+			OwnerAccountID:   nil,
+			TokenID:          &tokenID,
+			SerialNumbers:   []int64{},
+			AllSerials:       true,
+			DelegatingSpender: nil,
+		},
+	}
+	require.Equal(t, expectedTokenNftAllowances, tx.GetTokenNftAllowances())
+}
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**nitpick:** ‏Missing newline at end of file.